src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -514,7 +514,7 @@ private CodeProperty CreateProperty(string childIdentifier, string childType, st
         config.PropertiesPrefixToStrip.ForEach(x => propertyName = propertyName.Replace(x, string.Empty));
         var prop = new CodeProperty
         {
-            Name = propertyName,
+            Name = propertyName.ToCamelCase(),//ensure the name is camel cased to strip out any potential '-' characters
```

Comment:
assign the propertyName variable instead or otherwise the deserialization will fail because the serialization name won't be set properly at line 522

Reply:
Thanks for that!

Reply:
Resolved in latest update